### DISCUSSION TITLE
Inquiry Regarding YOLOv8 Model Configuration for Specific Categories and Accuracy Thresholds

### DISCUSSION THREAD

**Discussion by Nick11220** [ultralytics/ultralytics#6542] [Q&A]
Hi,

I hope this message finds you well. I am currently working with YOLOv8 for object detection, and I have a specific configuration requirement. In my model, I have six categories, but for the initial stage of prediction, I only want to focus on detecting four specific categories, excluding the other two.

Furthermore, I aim to set distinct accuracy thresholds for these categories during prediction. Specifically, I would like the first category to be detected with a confidence threshold of 40%, the second with 60%, and the third and fourth with a higher confidence threshold of 90%.

Could you kindly guide me on how to configure my YOLOv8 model to achieve this specific category selection and accuracy threshold setup?

Best regards

**Comment by pderrenger** [comment]
@Nick11220 hello,

To focus on detecting specific categories, you can use the `classes` parameter during prediction to filter results by class indices. For setting distinct confidence thresholds for each category, you would need to post-process the detection results since YOLOv8 does not natively support per-class confidence thresholds.

Here's a simplified example of how you might approach this:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/model.pt')

# Perform prediction
results = model.predict(source='path/to/your/image/or/video')

# Post-process results to apply per-class confidence thresholds
desired_classes = [0, 1, 2, 3]  # indices of the four categories you want to detect
thresholds = [0.4, 0.6, 0.9, 0.9]  # confidence thresholds for each of the four categories

filtered_boxes = []
for i, class_index in enumerate(desired_classes):
    # Filter detections for the current class
    class_detections = results.boxes[results.boxes.cls == class_index]
    # Apply the confidence threshold
    class_detections = class_detections[class_detections.conf >= thresholds[i]]
    filtered_boxes.append(class_detections)

# Now, 'filtered_boxes' contains detections for your four categories with applied thresholds
```

Please note that you'll need to adjust the indices in `desired_classes` to match the indices of the categories in your dataset. The `thresholds` list should correspond to these indices.

Remember, this is a simplified example, and you may need to adapt it to fit the exact structure of your results object and your specific use case.

Best regards!
